lib/dap/filter/simple.rb

```diff
@@ -171,6 +171,20 @@ def process(doc)
   end
 end
 
+class FilterFlatten
+  include Base
+  def process(doc)
+    self.opts.each_pair do |k,v|
+      if doc.has_key?(k) and doc[k].is_a?(Hash)
```

Comment:
Perhaps a `flatten_all` would also be nice to have to not have to know what levels you want to flatten beforehand...